### PR TITLE
[VLM] Extracted Input Embedder from VLM Pipeline

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
**Changes:**
- Extracted InputEmbedder from VLMPipeline class, which prepares inputs for LLM model
- InputEmbedder has backends per model - MiniCPM, Llava or Llava-Next
- Main VLMPipeline delegates inputs handling and chat history to InputEmbedder 
- InputEmbedder can be reused in CB later to enable Visual Language models serving

### CODE REVIEW

File: src/cpp/src/visual_language/input_embedder.cpp [hunk 1]
```diff
@@ -0,0 +1,899 @@
...
+                for (size_t l = 0; l < second.get_shape().at(2); ++l, ++k) {
+                    res_data[i * res_d_1 * res_d_2 + j * res_d_2 + k]
+                        = second_data[i * res_d_1 * second.get_shape().at(2) + j * second.get_shape().at(2) + l];
+                }
+            }
+        }
+        return res;
+    }
+
+    ov::Tensor concatenate_mid_dim(const ov::Tensor& first, const ov::Tensor& second) {
```
Comment (Wovchena) [inline src/cpp/src/visual_language/input_embedder.cpp:343]: It seems it's no longer used.

File: src/cpp/src/visual_language/pipeline.cpp [hunk 2]
```diff
@@ -768,29 +197,24 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
...
+        m_is_chat_conversation = false;
+        m_inputs_embedder->finish_chat();
+    }
 
     Tokenizer get_tokenizer() const {
         return m_tokenizer;
     }
 
     void set_chat_template(const std::string& new_template) {
+        OPENVINO_ASSERT(!m_is_chat_conversation, "Chat template cannot be changed once start_chat() is called. Please, finish current chat via finish_chat()");
```
Comment (Wovchena) [inline src/cpp/src/visual_language/pipeline.cpp:217]: That's no what docstrings state :)
```
    /// applies chat_template to input prompts. Calling start_chat()
    /// again or finish_chat() drops the memorized history.
```
Update it please.

**Review by Wovchena — CHANGES_REQUESTED** [review]
Accuracy test diverged

File: src/cpp/src/visual_language/input_embedder.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/cpp/src/visual_language/input_embedder.cpp]: dropped

**Review by Wovchena — APPROVED** [review]

**Pull request merged**